orchestrator/gravity_utils/src/error.rs

```diff
@@ -25,21 +24,21 @@ use tonic::Status;
 
 #[derive(Debug)]
 #[allow(clippy::large_enum_variant)]
-pub enum GravityError {
+pub enum GravityError<S: Signer> {
```

Comment:
Yeah, I don't like it either -- the culprit here is this nested error extraction: https://github.com/crypto-org-chain/gravity-bridge/blob/4369b5013716e5d06291178db8f5c23f9b466a7f/orchestrator/ethereum_gravity/src/utils.rs#L229 which also looks bad.

I don't know how important that extraction is, but one way to avoid the generic type in the error is to disable that error extraction (just return `None`) and just map those ethers's error to strings

Reply:
Or just create a variant for Signer error with `Box<dyn Error>` if we don't need check the error details.

Reply:
one alternative perhaps is to do some dynamic check of the type + an unsafe typecast

Reply:
Yeah. I too am not sure what is the best way as the underlying crate has generics in error type. Maybe we can push this for now and try and fix the generics issues in future PR?

Reply:
perhaps doesn't need to be unsafe: https://doc.rust-lang.org/std/any/trait.Any.html#method.downcast_ref

Reply:
@devashishdxt I tried to remove the generic type from the error type -- it bubbled up `'static` in many places (starting with the `From` implementation for those would have an error that it could not determine if `S` lived long enough).
And I tried to rewrite that error extraction using downcasts: https://github.com/crypto-org-chain/gravity-bridge/pull/106/files#diff-fc9f3265224358be9b523abf3169ba162ae0392f4bb8057386f1a36119ac92bcR239 not sure if correct though